lib/encoder.js

```diff
@@ -75,16 +76,23 @@ function parseTag(obj, tagName, codePage, codePageMapping, currentCodePage) {
 		throw new Error('Token for tag ' + tagSuffix + ' not found');
 	}
 
+	//create tag identifier
+	tagBuffer = Buffer.concat([ new Buffer([tagCode.token | (hasRealAttributes ? 128 : 0) | (hasContent? 64 : 0)]) ]);
+
 	if (tagCodePage !== currentCodePage) {
 		//switch code page
-		tagBuffer = new Buffer([SWITCH_PAGE_TOKEN, tagCodePage]);
+		tagBuffer = Buffer.concat([tagBuffer, new Buffer([SWITCH_PAGE_TOKEN, tagCodePage])])
```

Comment:
From my tests, I think it works as intended this way. Otherwise the tags are not being closed correctly.